app/assets/javascripts/discourse/app/components/post.gjs

```diff
@@ -326,301 +325,280 @@ export default class Post extends Component {
   }
 
   <template>
-    {{#if @cloaked}}
-      {{! temporary workaround to set the parent class using the modifier while the post stream widget is not modernized }}
-      {{! TODO (glimmer-post-stream): remove when the post stream is modernized and move the class to the outer div }}
-      <article
-        {{parentClass "post--cloaked cloaked-post" parentSelector="div"}}
-      ></article>
-    {{else}}
-      {{#let
-        (hash
-          post=@post
-          actions=(hash updateTopicPageQueryParams=@updateTopicPageQueryParams)
-          topicPageQueryParams=@topicPageQueryParams
+    <div
+      ...attributes
+      class={{unless
+        @cloaked
+        (concatClass
+          "topic-post"
+          "clearfix"
+          (unless this.site.mobileView "post--sticky-avatar sticky-avatar")
+          (if this.staged "post--staged staged")
+          (if @selected "post--selected selected")
+          (if @post.topicOwner "post--topic-owner topic-owner")
+          (if this.isFromCurrentUser "post--current-user current-user-post")
+          (if
+            @post.group_moderator "post--category-moderator category-moderator"
+          )
+          (if @post.hidden "post--hidden post-hidden")
+          (if @post.deleted "post--deleted deleted")
+          (if
+            @post.primary_group_name
+            (concatClass
+              (concat "post--group-" @post.primary_group_name)
+              (concat "group-" @post.primary_group_name)
+            )
+          )
+          (if @post.wiki "post--wiki wiki")
+          (if @post.isWhisper "post--whisper whisper")
+          (if
+            (or @post.isModeratorAction (and @post.isWarning @post.firstPost))
+            "post--moderator moderator"
+            "post--regular regular"
+          )
+          (if @post.user_suspended "post--user-suspended user-suspended")
+          this.additionalClasses
         )
-        as |postOutletArgs|
       }}
-        <PluginOutlet @name="post-article" @outletArgs={{postOutletArgs}}>
-          <article
-            {{parentClass
-              (concatClass
-                "topic-post"
-                "clearfix"
-                (if this.staged "post--staged staged")
-                (if @selected "post--selected selected")
-                (if @post.topicOwner "post--topic-owner topic-owner")
+      data-post-number={{@post.post_number}}
+    >
+      {{#unless @cloaked}}
+        {{#let
+          (hash
```

Comment:
To avoid the Plugin Outlets rerendering unnecessarily:
```suggestion
          (lazyHash
```

and 

```
import lazyHash from "discourse/helpers/lazy-hash";
```

(this is normally linted for, but the rule didn't catch this because the hash is being built separately from the outlet)